tests/monitoring_processes/test_monitoring_critical_processes.py

```diff
@@ -0,0 +1,543 @@
+            else:
+                containers_in_namespaces[container_name].append("host")
+            logger.info("The configuration of container '{}' in `FEATURE` table was retrieved.".format(container_name))
+
+    return containers_in_namespaces
+
+
+def disable_containers_autorestart(duthost, containers_in_namespaces):
+    """Disables the autorestart of enabled containers.
+
+    Args:
+        duthost: Hostname of DUT.
+        containers_in_namespaces: A dictionary where keys are container names and
+        values are lists which contains ids of namespaces this container should reside in.
+
+    Returns:
+        None.
+    """
+    for container_name in containers_in_namespaces.keys():
+        logger.info("Disabling the autorestart of container '{}'.".format(container_name))
+        command_disable_autorestart = "sudo config feature autorestart {} disabled".format(container_name)
+        command_output = duthost.shell(command_disable_autorestart)
+        exit_code = command_output["rc"]
+        pytest_assert(exit_code == 0, "Failed to disable the autorestart of container '{}'".format(container_name))
+        logger.info("The autorestart of container '{}' was disabled.".format(container_name))
+
+
+def get_group_program_info(duthost, container_name, group_name):
+    """Gets program names, running status and their pids by analyzing the command
+       output of "docker exec <container_name> supervisorctl status". Program name
+       at here represents a program which is part of group <group_name>
+
+    Args:
+        duthost: Hostname of DUT.
+        container_name: A string shows container name.
+        program_name: A string shows process name.
+
+    Returns:
+        A dictionary where keys are the program names and values are their running
+        status and pids.
+    """
+    group_program_info = defaultdict(list)
+    program_name = None
+    program_status = None
+    program_pid = -1
+
+    program_list = duthost.shell("docker exec {} supervisorctl status".format(container_name), module_ignore_errors=True)
+    for program_info in program_list["stdout_lines"]:
+        if program_info.find(group_name) != -1:
+            program_name = program_info.split()[0].split(':')[1].strip()
+            program_status = program_info.split()[1].strip()
+            if program_status in ["EXITED", "STOPPED", "STARTING"]:
+                program_pid = -1
+            else:
+                program_pid = int(program_info.split()[3].strip(','))
+
+            group_program_info[program_name].append(program_status)
+            group_program_info[program_name].append(program_pid)
+
+            if program_pid != -1:
+                logger.info("Found program '{}' in the '{}' state with pid {}"
+                            .format(program_name, program_status, program_pid))
+
+    return group_program_info
+
+
+def get_program_info(duthost, container_name, program_name):
```

Comment:
Should this function be a shared helper function?

Reply:
Moved this function to `tests/common/helers/dut_utils.py`.